src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -174,6 +183,12 @@ public class CompactionManager implements CompactionManagerMBean, ICompactionMan
     final Multiset<ColumnFamilyStore> compactingCF = ConcurrentHashMultiset.create();
 
     public final ActiveCompactions active = new ActiveCompactions();
+    private final Cache<TimeUUID, OperationInfo> operations = CacheBuilder.newBuilder()
```

Comment:
I wanted to say that his should be "space-bound" instead of "time-bound" but I noticed that you actually did both which is great. If somebody goes crazy and they cleanup like mad then without the limit on size this would grow and grow without any control.

I would just like to know based on what you did it like 7 days and 300 entries. Is this empirical based on your workload? I think revisiting the decision about the figures here would be nice.

Also, does the cache preserve any insertion order? If you do operations.toMap(), do I get entries without any particular order in the virtual table? I would like them to be order the same way every single time.

Reply:
I tend to use "space-bound" only in favor of both options altogether. Both of them are configurable via the system settings, so it should be no problem to customize them for any type of deployment. Based on the SLA we have and my previous experience it should be fine having 7d and 300 entries, but of course we can update these numbers.

The insert order is timestamp-UUID, which is a primary key for the virtual table we are trying to add, so with a default query it will be sorted naturally. I've checked the caching API and haven't found any sorting capabilities, but I don't think it should matter for such a small table.